guides/v2.1/howdoi/checkout/checkout-add-custom-carrier.md

```diff
@@ -0,0 +1,297 @@
+                </field>
+            </group>
+        </section>
+    </system>
+</config>
+```
+
+### Source code of `app/code/Vendor/CustomShipping/etc/config.xml`
+
+File specifies default values for custom shipping module options and shipping module model - `Vendor\CustomShipping\Model\Carrier\Customshipping`.
+
+```xml
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Store:etc/config.xsd">
+    <default>
+        <carriers>
+            <customshipping>
+                <active>0</active>
+                <title>Custom Shipping Title</title>
+                <name>Custom Shipping Method Name</name>
+                <shipping_cost>10</shipping_cost>
+                <sallowspecific>0</sallowspecific>
+                <sort_order>999</sort_order>
+                <model>Vendor\CustomShipping\Model\Carrier\Customshipping</model>
+            </customshipping>
+        </carriers>
+    </default>
+</config>
+```
+
+## Step 3: Create configuration file {#carrier-model}
+
+### Source code of `app/code/Vendor/CustomShipping/Model/Carrier/Customshipping.php`
+
+```php
+<?php
+
+namespace Vendor\CustomShipping\Model\Carrier;
+
+use Magento\Quote\Model\Quote\Address\RateRequest;
+use Magento\Shipping\Model\Carrier\AbstractCarrier;
+use Magento\Shipping\Model\Carrier\CarrierInterface;
+
+/**
+ * Custom shipping model
+ */
+class Customshipping extends AbstractCarrier implements CarrierInterface
+{
+    /**
+     * @var string
+     */
+    protected $_code = 'customshipping';
+
+    /**
+     * @var bool
+     */
+    protected $_isFixed = true;
+
+    /**
+     * @var \Magento\Shipping\Model\Rate\ResultFactory
+     */
+    protected $_rateResultFactory;
+
+    /**
+     * @var \Magento\Quote\Model\Quote\Address\RateResult\MethodFactory
+     */
+    protected $_rateMethodFactory;
+
+    /**
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     * @param \Magento\Quote\Model\Quote\Address\RateResult\ErrorFactory $rateErrorFactory
+     * @param \Psr\Log\LoggerInterface $logger
+     * @param \Magento\Shipping\Model\Rate\ResultFactory $rateResultFactory
+     * @param \Magento\Quote\Model\Quote\Address\RateResult\MethodFactory $rateMethodFactory
+     * @param array $data
+     */
+    public function __construct(
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Quote\Model\Quote\Address\RateResult\ErrorFactory $rateErrorFactory,
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Shipping\Model\Rate\ResultFactory $rateResultFactory,
+        \Magento\Quote\Model\Quote\Address\RateResult\MethodFactory $rateMethodFactory,
+        array $data = []
+    ) {
+        $this->_rateResultFactory = $rateResultFactory;
+        $this->_rateMethodFactory = $rateMethodFactory;
+        parent::__construct($scopeConfig, $rateErrorFactory, $logger, $data);
```

Comment:
Please, put the parent constructor invocation to the very beginning of the constructor. 
```suggestion
       parent::__construct($scopeConfig, $rateErrorFactory, $logger, $data);
       $this->_rateResultFactory = $rateResultFactory;
       $this->_rateMethodFactory = $rateMethodFactory;
```

Reply:
**done**

Reply:
**done**